server/tests/experiments/test_preloading.py

```diff
@@ -1,9 +1,14 @@
+import pytest
+
 from athenian.api.controllers.features.entries import make_calculator
 from athenian.api.experiments.preloading import MetricEntriesCalculator
 
 
+# TODO: this fails because precomputed data are not loaded for unittest
```

Comment:
Tbh I'd rather not do this in this PR, I don't even know how much work it could require.

Reply:
What's happening here?

Reply:
The problem is that the `is_ready_for` checks for the data being indeed preloaded to avoid querying an empty df. This check is implemented by checking whether there's the corresponding shard loaded. The problem is that in the unittests we only load `metadata`, but not `precomputed`, so this check always fails. We need to also load for tests the `precomputed` data.

Reply:
In production those data would be ready since we don't open access until the heater runs.

Reply:
We've got the `app` fixture. We can extend it with `is_preloading_enabled` and await the preloader if it is True. Should be easy.

Reply:
I think I'm missing something, the problem is not the preloader being ready, but that we don't have any data in the `precomputed` database. So AFAIU to solve this we should either create the fixture and load those the same way we do for `metadata` or run the heater on top of the loaded `metadata` data.

Reply:
Got it 👍